state/errors.go

```diff
@@ -43,9 +43,9 @@ var (
 	// ongoing batch are not in the same order as the transactions stored in the
 	// database for the same batch.
 	ErrOutOfOrderProcessedTx = errors.New("the processed transactions are not in the same order as the stored transactions")
-	// ErrInsufficientFunds is returned if the total cost of executing a transaction
-	// is higher than the balance of the user's account.
-	ErrInsufficientFunds = errors.New("insufficient funds for gas * price + value")
+	// ErrInsufficientFundsForTransfer is returned if the transaction sender doesn't
+	// have enough funds for transfer(topmost call only).
+	ErrInsufficientFundsForTransfer = errors.New("insufficient funds for transfer")
```

Comment:
I didn't notice it in the other review but what about using transaction instead of transfer? Transfer is tipically used for ether txs